src/Pokemon.java

```diff
@@ -0,0 +1,175 @@
+import java.util.List;
+
+public abstract class Pokemon {
+
+    private final String name;
+    private int level;
+    private int hp;
+    private final String sound;
+    private final String food;
+    private String type;
+
+    public Pokemon(String name, int hp, int level, String food, String sound) {
+        this.food = food;
+        this.hp = hp;
+        this.level = level;
+        this.name = name;
+        this.sound = sound;
+    }
+
+    public String getFood() {
+        return food;
+    }
+
+
+    public int getHp() {
+        return hp;
+    }
+
+    public void setHp(int hp) {
+        this.hp = hp;
+    }
+
+    public int getLevel() {
+        return level;
+    }
+
+    public void setLevel(int level) {
+        this.level = level;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+
+    public String getSound() {
+        return sound;
+    }
+
+
+    public String getType() {
+        return type;
+    }
+
+    public void setType(String type) {
+        this.type = type;
+    }
+
+    abstract List<String> getAttacks();
+
+
+    public void determineDmgOutput(String type, Pokemon enemy) {
+        switch (type) {
+            case "fire":
+                fireDmgOutput(enemy);
+                break;
+            case "water":
+                waterDmgOutput(enemy);
+                break;
+            case "electric":
+                electricDmgOutput(enemy);
+                break;
+            case "grass":
+                grassDmgOutput(enemy);
+                break;
+        }
+    }
+
+    public void electricDmgOutput(Pokemon enemy) {
```

Comment:
Deze specifieke type-output methodes, kun je beter `private` maken, want je gebruikt ze niet buiten de Pokemon klassen. In de ElectricPokemon klasse roep je niet deze methode aan, maar de `determineDamageOutput` methode.